pandas/tests/scalar/test_interval_new.py

```diff
@@ -0,0 +1,294 @@
+            'right': {'right': True, 'left': False, 'both': False, 'neither': True},
+            'left': {'right': False, 'left': True, 'both': False, 'neither': True},
+            'both': {'right': True, 'left': True, 'both': True, 'neither': True},
+            'neither': {'right': False, 'left': False, 'both': False, 'neither': True}
+        }
+
+        ivl = Interval(1, 3, closed=ivl_side)
+        other = Interval(1, 3, closed=other_side)
+        assert ivl.covers(other) == should_cover[ivl_side][other_side]
+
+    @pytest.mark.parametrize("ivl_side", ['right', 'left', 'both', 'neither'])
+    @pytest.mark.parametrize("other_side", ['right', 'left', 'both', 'neither'])
+    @pytest.mark.parametrize("ivl_range", [(1, 3), (-1, 1), (3, 5)])
+    def test_interval_overlaps_interval(self, ivl_side, other_side, ivl_range):
+
+        # class Interval:
+        #     def overlaps(self, other: Interval) -> bool
+
+        assert Interval(1, 3).overlaps(Interval(1.5, 2.5))
+        assert Interval(1, 3).overlaps(Interval(1, 2))
+        assert Interval(1, 3).overlaps(Interval(2, 3))
+        assert Interval(1, 3).overlaps(Interval(0.5, 2.5))
+        assert Interval(1, 3).overlaps(Interval(1.5, 3.5))
+
+        assert not Interval(1, 3).overlaps(Interval(-1, 1))
+        assert not Interval(1, 3).overlaps(Interval(3, 5))
+
+        ivl = Interval(*ivl_range, closed=ivl_side)
+        other = Interval(1, 3, closed=other_side)
+
+        should_overlap = {
+            # idx_side:
+            #   ivl_side: {ivl_range: expected_result}
+            'right': {
+                'right': {(-1, 1): False, (1, 3): True, (3, 5): False},
+                'left':{(-1, 1): False, (1, 3): True, (3, 5): True},
+                'both': {(-1, 1): False, (1, 3): True, (3, 5): True},
+                'neither': {(-1, 1): False, (1, 3): True, (3, 5): False},
+            },
+            'left': {
+                'right': {(-1, 1): True, (1, 3): True, (3, 5): False},
+                'left':{(-1, 1): False, (1, 3): True, (3, 5): False},
+                'both': {(-1, 1): True, (1, 3): True, (3, 5): False},
+                'neither': {(-1, 1): False, (1, 3): True, (3, 5): False},
+            },
+            'both': {
+                'right': {(-1, 1): True, (1, 3): True, (3, 5): False},
+                'left':{(-1, 1): False, (1, 3): True, (3, 5): True},
+                'both': {(-1, 1): True, (1, 3): True, (3, 5): True},
+                'neither': {(-1, 1): False, (1, 3): True, (3, 5): False},
+            },
+            'neither': {
+                'right': {(-1, 1): False, (1, 3): True, (3, 5): False},
+                'left':{(-1, 1): False, (1, 3): True, (3, 5): False},
+                'both': {(-1, 1): False, (1, 3): True, (3, 5): False},
+                'neither': {(-1, 1): False, (1, 3): True, (3, 5): False},
+            }
+        }
+
+        assert ivl.overlaps(other) == should_overlap[other_side][ivl_side][ivl_range] == other.overlaps(ivl)
```

Comment:
all of the tests from here on down should go in: ``pandas/tests/indexes/interval/test_overlaps.py`` (new file)

Reply:
@jreback do you still want to see this?